### ISSUE TITLE
Empty String goes through array validation

### ISSUE THREAD

**Issue by rao-f** [trenskow/isvalid#18]
Hello, I was trying to validate a type of Array and it seems to treat empty string as a valid array in the schema validation. 

the schema is like:

``` javascript
var mySchema = {
  type: Object,
  unknownKeys: 'remove',
  schema: {
    anArrayOfObjects: {
      type: Array,
      schema: {
        type: Object,
        schema: {
          propOne: String,
          propTwo: Boolean
        }
      }
    }
  }
};
```

While the test value is:

``` javascript
var test = {
  name: 'empty string',
  anArrayOfObjects: ''
};
```

After using isvalid(test, mySchema, func), the result shows the test value is valid. Shouldn't it throw out a validation error? 

Thanks

**Comment by trenskow** [comment]
It should throw a type error, yes. I'll investigate...1

**Comment by rao-f** [comment]
In the code, values like **'', false, 0 an NaN** will go through validation as well. It seems that the if condition **on line 79 in validate.js** will skip Array type checking on those values: 
```javascript
var validateArray = function(data, schema, fn, keyPath, options) {

	if (data) {
          ...
        }
        return validateCustom(data, schema, fn, keyPath, options);
}
```
I can add a customized function in the schema to throw out type errors on those values listed above, but I'm wondering if such values can be checked without customized functions because it is a little confusing :-p. 

Thank you

**Comment by trenskow** [comment]
This is most definitely a bug. I'll fix it this weekend. Until then, you can use a custom.

BTW. Nice catch!

This must be the guilty line.
https://github.com/trenskow/isvalid/blob/master/lib/validate.js#L81

**Comment by rao-f** [comment]
Great. Thanks!

**Comment by trenskow** [comment]
The weekend came and went. Sorry! I'll look at this again.
